shared_model/backend/protobuf/block.hpp

```diff
@@ -117,6 +117,13 @@ namespace shared_model {
         return *payload_blob_;
       }
 
+      virtual const HashType &hash() const override {
```

Comment:
need to write documentation about purposes of reworking hash for the block.
And maybe better to move payload to hashable and rework hash here?

Reply:
The problem is that classes derived from Hashable may not have payload. And payload_blob_ != blob_ in Block.

Reply:
What do you mean by "may not have payload"? Block has it

Reply:
Old model block calculate hash from payload, shared model calculate from blob. The problem is that after conversion from old to new and vice versa hashes are not equal.
We can remove this function and keep derived hash from blob if we wouldn't convert from old to new model. Or move it to hashable, or keep it here